icons/vault.svg

```diff
@@ -0,0 +1,22 @@
+<svg
+  xmlns="http://www.w3.org/2000/svg"
+  width="24"
+  height="24"
+  viewBox="0 0 24 24"
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <rect width="18" height="18" x="3" y="3" rx="2" />
+  <circle cx="7.5" cy="7.5" r=".5" />
```

Comment:
Still not a fan of the .5 radius circles, in Figma they look awful:
![image](https://github.com/lucide-icons/lucide/assets/17746067/b2cbd9f8-0707-4f08-8e71-b7209f9e033c)

Maybe, like `safe`, this could also do with some legs to make it more legible?